docs/fundamentals/database-collection.txt

```diff
@@ -0,0 +1,164 @@
+.. _laravel-db-coll:
+
+=========================
+Databases and Collections
+=========================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: php framework, odm
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use {+odm-short+} to access
+and manage MongoDB databases and collections.
+
+MongoDB organizes data in a hierarchical structure. A MongoDB
+deployment contains one or more **databases**, and each database
+contains one or more **collections**. In each collection, MongoDB stores
+data as **documents** that contain field-and-value pairs.
+
+To learn more about the document data format,
+see :manual:`Documents </core/document/>` in the Server manual.
+
+.. _laravel-access-db:
+
+Access a Database
+-----------------
+
+You can access a database by specifying a database connection in your
+application's ``config/database.php`` file. The ``connections`` property
+in this file stores all of your database connection information, such as
+your connection string, database name, and optionally, authentication
+details. After you specify a database connection, you can perform
+database-level operations and access collections that the database
+contains.
+
+If you set the ``database`` property in a connection to the name of a
+nonexistent database, Laravel still makes a valid connection. When you
+insert any data into a collection in the database, the server creates it
+automatically.
+
+The following example shows how to set a default database connection and
+create a database connection to the ``animals`` database in the
+``config/database.php`` file:
+
+.. code-block:: php
+   :emphasize-lines: 8
+   
+   'default' => env('DB_CONNECTION', 'mongodb'),
+
+   'connections' => [
+
+       'mongodb' => [
+           'driver' => 'mongodb',
+           'dsn' => env('DB_URI', '<connection string>'),
```

Comment:
The `config/dabase.php` file is committed in the git repository of each project. The MongoDB DSN usually contains credentials, it should be in the `.env` file.
The 2nd argument of `env()` is the default value. Typically `mongodb://localhost:27017` for the MongoDB DSN.

The application can contain a `.env.example` but in local the developers must create the file `.env`. For production, there is many ways to set the environment variables, from the `.env` file to server or container config.

Reply:
I believe this is content that will be better suited for the Connection Guide

Reply:
This placeholder is wrong. It's better removing the fallback value then.
```suggestion
           'dsn' => env('DB_URI'),
```

Reply:
This is wrong. When a "dsn" is provided, the database name is read from the "dsn", and the  "database" config is ignored.


Your example should look like this, so we don't bother with env vars.
```php

       'mongodb' => [
           'driver' => 'mongodb',
           'dsn' => 'mongodb://localhost:27017/animals,
        ],
```

Reply:
> When a "dsn" is provided, the database name is read from the "dsn", and the "database" config is ignored.

Outside the scope of this PR, but is that something we should consider changing in some future release. The "auth database" component in the DSN should ideally never be specified since it was made obsolete by the `authSource` URI option (as I explained in https://github.com/symfony-cli/symfony-cli/pull/458#issuecomment-2025269218).